src/block/ChiseledBookshelf.php

```diff
@@ -97,8 +97,13 @@ public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player
 			return false;
 		}
 
+		$x = Facing::axis($face) === Axis::X ? $clickVector->getZ() : $clickVector->getX();
+		$x = match($face){
```

Comment:
The more logical way would be to check `Facing::isPositive(Facing::rotateY($face, true))`, in which case `1 - $x` is required. It took me a while to make full sense of why this applies to north and east, when the real issue concerns the axis the bookshelf's face is parallel to.